backend/dataall/modules/dataset_sharing/api/resolvers.py

```diff
@@ -218,6 +218,25 @@ def list_shareable_objects(
     )
 
 
+def resolve_shared_database_name(
+    context: Context, source
+):
+    if not source:
+        return None
+    old_shared_db_name = (source.GlueDatabaseName + '_shared_' + source.shareUri)[:254]
+    with context.engine.scoped_session() as session:
+        share = ShareObjectService.get_share_object_in_environment(uri=source.environmentUri, shareUri=source.shareUri)
```

Comment:
Why did we get the share item from this new method instead of getting it from get_share_object() ? Did we just want to confirm if the user has the permissions to access that env ?

Reply:
believe it is exactly for that reason because of decorator `@has_resource_permission(GET_ENVIRONMENT)` but will leave it open for @dlpzx  to confirm

Reply:
Thanks @noah-paige , @dlpzx for the clarification